drivers/wifi/esp8266/wifi_esp8266.c

```diff
@@ -0,0 +1,1570 @@
+	struct net_buf		*pkt_buf;
+	int			ret;
+	char tmp[16];
+	/** semaphore */
+	int connected;
+	int read_error;
+};
+
+struct esp8266_data {
+	struct net_if *iface;
+	struct in_addr ip;
+	struct in_addr gw;
+	struct in_addr netmask;
+	u8_t mac[6];
+
+	/* modem stuff */
+	struct mdm_receiver_context mdm_ctx;
+	int last_error;
+
+	struct k_delayed_work ip_addr_work;
+
+	/* semaphores */
+	struct k_sem response_sem;
+
+	/* max sockets */
+	struct socket_data socket_data[ESP8266_MAX_CONNECTIONS];
+	u8_t sock_map;
+
+	/* wifi scan callbacks */
+	scan_result_cb_t wifi_scan_cb;
+
+	int sending_data;
+
+	int data_id;
+	int data_len;
+
+	/* delayed work requests */
+
+	/* device management */
+	struct device *uart_dev;
+	struct device *gpio_dev;
+};
+
+static struct esp8266_data foo_data;
+
+#define MDM_CMD_TIMEOUT		K_SECONDS(5)
+
+static inline void _hexdump(const u8_t *packet, size_t length)
+{
+	char output[sizeof("xxxxyyyy xxxxyyyy")];
+	int n = 0, k = 0;
+	u8_t byte;
+
+	while (length--) {
+		if (n % 16 == 0) {
+			printk(" %08X ", n);
+		}
+
+		byte = *packet++;
+
+		printk("%02X ", byte);
+
+		if (byte < 0x20 || byte > 0x7f) {
+			output[k++] = '.';
+		} else {
+			output[k++] = byte;
+		}
+
+		n++;
+		if (n % 8 == 0) {
+			if (n % 16 == 0) {
+				output[k] = '\0';
+				printk(" [%s]\n", output);
+				k = 0;
+			} else {
+				printk(" ");
+			}
+		}
+	}
+
+	if (n % 16) {
+		int i;
+
+		output[k] = '\0';
+
+		for (i = 0; i < (16 - (n % 16)); i++) {
+			printk("   ");
+		}
+
+		if ((n % 16) < 8) {
+			printk(" "); /* one extra delimiter after 8 chars */
+		}
+
+		printk(" [%s]\n", output);
+	}
+}
+
+
+/* Send an AT command with a series of response handlers */
+static int send_at_cmd(struct socket_data *sock,
+			const u8_t *data, int timeout)
+{
+	int ret;
+
+	foo_data.last_error = 0;
+
+	k_mutex_lock(&dev_mutex, K_FOREVER);
+	k_sem_reset(&foo_data.response_sem);
+	mdm_receiver_send(&foo_data.mdm_ctx, data, strlen(data));
+	mdm_receiver_send(&foo_data.mdm_ctx, "\r\n", 2);
+	k_mutex_unlock(&dev_mutex);
+
+	if (timeout == K_NO_WAIT) {
+		return 0;
+	}
+
+	ret = k_sem_take(&foo_data.response_sem, timeout);
+
+	if (ret == 0) {
+		ret = foo_data.last_error;
+	} else if (ret == -EAGAIN) {
+		ret = -ETIMEDOUT;
+	}
+
+	return ret;
+}
+
+static int net_buf_find_next_delimiter(struct net_buf *buf, const u8_t d,
+	size_t index)
+{
+	struct net_buf *frag = buf;
+	u16_t offset = 0;
+	u16_t pos = 0;
+
+	while (frag && index) {
+		if (frag->len > index) {
+			offset += index;
+			pos += index;
+			break;
+		} else {
+			index -= frag->len;
+			pos += frag->len;
+			offset = 0;
+			if (!frag->frags) {
+				return -1;
+			}
+			frag = frag->frags;
+		}
+	}
+
+	while(*(frag->data + offset) != d) {
+		if (offset == frag->len) {
+			if (!frag->frags) {
+				return -1;
+			}
+			frag = frag->frags;
+			offset = 0;
+		} else {
+			offset++;
+		}
+		pos++;
+	}
+
+	return pos;
+}
+
+static int net_buf_ncmp(struct net_buf *buf, const u8_t *s2, size_t n)
```

Comment:
Please rename this to some other name than "net_buf_", I was trying to search this function from buf.h as the naming would indicate it would be found there.